src/core/search/range_tree.cc

```diff
@@ -221,37 +243,44 @@ TODO: we can optimize this case by splitting to three blocks:
  - empty right block with range [std::nextafter(m, +inf), r)
 */
 void RangeTree::SplitBlock(Map::iterator it) {
-  const RangeNumber l = it->first;
+  double l = it->first;
 
   auto split_result = Split(std::move(it->second));
 
-  const RangeNumber m = split_result.median;
+  const double m = split_result.median;
   DCHECK(!split_result.right.Empty());
 
   entries_.erase(it);
+  stats_.splits++;
+
+  // Insert left block if its not empty, but always keep left infinity bound
+  if (!split_result.left.Empty() || std::isinf(l)) {
```

Comment:
in what cases `std::isinf(l)` is true?

Reply:
ok, I had to read that std::inf works on negative infinity as well. Maybe worth adding a comment that you keep left `-inf bound`